lib/elixir_internal_certificate_worker/scraper/user_search_worker.ex

```diff
@@ -0,0 +1,75 @@
+defmodule ElixirInternalCertificateWorker.Scraper.UserSearchWorker do
+  use Oban.Worker,
+    queue: :user_search,
+    max_attempts: 3,
+    unique: [period: 30]
+
+  alias ElixirInternalCertificate.Repo
+  alias ElixirInternalCertificate.Scraper.{ScraperEngine, Scrapers}
+  alias ElixirInternalCertificateWorker.Scraper.HtmlParsingHelper
+
+  @max_attempts 3
+
+  @impl Oban.Worker
+  def perform(%Oban.Job{args: %{"user_search_id" => user_search_id}, attempt: @max_attempts}) do
+    {:ok, _} = update_user_search(user_search_id, :failed)
+
+    {:error, "max attempts reached, attempt: #{@max_attempts}"}
+  end
+
+  @impl Oban.Worker
+  def perform(%Oban.Job{args: %{"user_search_id" => user_search_id}}) do
+    {_, searching_keyword} = update_user_search(user_search_id, :in_progress)
+
+    {:ok, html_response} = ScraperEngine.get_html(searching_keyword.keyword)
+
+    {:ok, search_result} = HtmlParsingHelper.parsing(html_response)
+
+    Repo.transaction(fn ->
+      {:ok, search_result_record} =
+        Scrapers.saving_search_result(Map.put(search_result, :user_search_id, user_search_id))
+
+      {:ok} = saving_url_results(search_result, search_result_record.id)
+
+      {:ok, _} = update_user_search(user_search_id, :success)
```

Comment:
incase the action is failed, will we update the user_search status to `failed` 🤔

Reply:
@andyduong1920 this block is a transaction block, so whenever it falls into ```failed``` status, the transaction closes before needing of handling. Besides, it should be updated to ```failed``` in the last attempt is suited to my expectations. I'm I doing that right?